nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -1235,6 +1235,10 @@ pub enum OmicronZoneType {
     Nexus {
         /// The address at which the internal nexus server is reachable.
         internal_address: SocketAddrV6,
+        /// The port at which the internal lockstep server is reachable. This
+        /// shares the same IP address with `internal_address`.
+        #[serde(default = "default_nexus_lockstep_port")]
```

Comment:
I think we've usually avoided this sort of thing.  I guess to avoid it you'd need to define a separate struct here with a `From` impl, which may not be worth it.  But can we at least figure out when it would be safe to remove this (if ever?), document that, and file an issue?